### ISSUE TITLE
error: use of undeclared identifier 'HomeDir'

### ISSUE THREAD

**Issue by andreas-bulling** [neomutt/homebrew-neomutt#31]
Hi,

I've tried to update to the latest HEAD version (HEAD-99485dc) but the compile fails with the following error message:

```
==> make
Last 15 lines from /Users/andreas/Library/Logs/Homebrew/neomutt/02.make:
mv -f .deps/socket.Tpo .deps/socket.Po
clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT tunnel.o -MD -MP -MF .deps/tunnel.Tpo -c -o tunnel.o tunnel.c
clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT sasl.o -MD -MP -MF .deps/sasl.Tpo -c -o sasl.o sasl.c
clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT ssl.o -MD -MP -MF .deps/ssl.Tpo -c -o ssl.o ssl.c
mv -f .deps/tunnel.Tpo .deps/tunnel.Po
ssl.c:619:56: error: use of undeclared identifier 'HomeDir'
    snprintf(path, sizeof(path), "%s/.entropy", NONULL(HomeDir));
                                                       ^
ssl.c:619:56: error: use of undeclared identifier 'HomeDir'
mv -f .deps/sasl.Tpo .deps/sasl.Po
2 errors generated.
make[2]: *** [ssl.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/neomutt/homebrew-neomutt/issues

These open issues may also help:
Can't build neomutt with libncurses https://github.com/neomutt/homebrew-neomutt/issues/28
the --disable-nls line in the formula breaks head https://github.com/neomutt/homebrew-neomutt/issues/24
LUA_VERSION not found https://github.com/neomutt/homebrew-neomutt/issues/23
Add support for --with-lmdb. Make TC optional. https://github.com/neomutt/homebrew-neomutt/pull/19
```

openssl 1.0.2l installed via homebrew

**Comment by ThomasAdam** [comment]
On Sun, Oct 29, 2017 at 08:59:05PM +0000, andreas-bulling wrote:
> Hi,
> 
> I've tried to update to the latest HEAD version (HEAD-99485dc) but the compile fails with the following error message:
> 
> ```
> ==> make
> Last 15 lines from /Users/andreas/Library/Logs/Homebrew/neomutt/02.make:
> mv -f .deps/socket.Tpo .deps/socket.Po
> clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT tunnel.o -MD -MP -MF .deps/tunnel.Tpo -c -o tunnel.o tunnel.c
> clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT sasl.o -MD -MP -MF .deps/sasl.Tpo -c -o sasl.o sasl.c
> clang -DHAVE_CONFIG_H -I. -I..  -I.. -I/usr/local/opt/openssl/include  -Wall -pedantic -g -O2 -MT ssl.o -MD -MP -MF .deps/ssl.Tpo -c -o ssl.o ssl.c
> mv -f .deps/tunnel.Tpo .deps/tunnel.Po
> ssl.c:619:56: error: use of undeclared identifier 'HomeDir'
>     snprintf(path, sizeof(path), "%s/.entropy", NONULL(HomeDir));
>                                                        ^
> ssl.c:619:56: error: use of undeclared identifier 'HomeDir'
> mv -f .deps/sasl.Tpo .deps/sasl.Po
> 2 errors generated.
> make[2]: *** [ssl.o] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [all-recursive] Error 1
> make: *** [all] Error 2

This looks to be the problem:

diff --git a/conn/ssl.c b/conn/ssl.c
index 5aeffed51..fe3f99a64 100644
--- a/conn/ssl.c
+++ b/conn/ssl.c
@@ -70,6 +70,7 @@
 #include "options.h"
 #include "protos.h"
 #include "socket.h"
+#include "globals.h"
 
 /* Just in case OpenSSL doesn't define DEVRANDOM */
 #ifndef DEVRANDOM

I'm surprised the TravisCI builds didn't pick this up.

-- Thomas Adam

**Comment by alexpearce** [comment]
Yeah the CI should have seen this, strange.

I think the problem was fixed by neomutt/neomutt@18bf10e.
